inst/include/foo_dist_stan.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef testproject_foo_dist_stan_hpp
+#define testproject_foo_dist_stan_hpp 1
+
+#include "foo_dist.hpp"
+#include <stan/math/rev/core.hpp>
+
+// plain function
+// NOTE: for some reason doesn't work with _lpdf!
+double foo_dist_lpdfi(const double& y, const double& mu,
```

Comment:
TODO (neerajen) look into this

Reply:
That comment should have been removed...`_lpdf` (or whatever else) works fine as long as you include the `template <...> typename boost::math::tools::promote_args<...>` function signature in the hpp file, in addition to the `stan::math::var` functions.